src/status_im/models/wallet.cljs

```diff
@@ -51,3 +57,109 @@
 (defn edit-value
   [key value {:keys [db]}]
   {:db (update-in db [:wallet :edit] build-edit key value)})
+
+;; DAPP TRANSACTION -> SEND TRANSACTION
+(defn prepare-dapp-transaction [{{:keys [id method params]} :payload :as queued-transaction} contacts]
+  (let [{:keys [to value data gas gasPrice nonce]} (first params)
+        contact (get contacts (utils.hex/normalize-hex to))]
+    (cond-> {:id               (str id)
+             :to-name          (or (when (nil? to)
+                                     (i18n/label :t/new-contract))
+                                   contact)
+             :symbol           :ETH
+             :method           method
+             :dapp-transaction queued-transaction
+
+             :to               to
+             ;; TODO (andrey) why don't we apply money/to-decimal for amount
+             :amount           (money/bignumber (or value 0))
+             :gas              (when (seq gas)
+                                 (money/bignumber (money/to-decimal gas)))
+             :gas-price        (when (seq gasPrice)
+                                 (money/bignumber (money/to-decimal gasPrice)))
+             :data             data}
+      nonce
+      (assoc :nonce nonce))))
+
+;; SEND TRANSACTION -> RPC TRANSACTION
+(defn prepare-send-transaction [from {:keys [amount to gas gas-price data nonce]}]
+  (cond-> {:from      (ethereum/normalized-address from)
+           :to        (ethereum/normalized-address to)
+           :value     (ethereum/int->hex amount)
+           :gas       (ethereum/int->hex gas)
+           :gas-price (ethereum/int->hex gas-price)}
+    data
+    (assoc :data data)
+    nonce
+    (assoc :nonce nonce)))
+
+(defn normalize-sign-message-params [params]
+  (let [first_param           (first params)
+        second_param          (second params)
+        first-param-address?  (ethereum/address? first_param)
+        second-param-address? (ethereum/address? second_param)]
+    (when (or first-param-address? second-param-address?)
```

Comment:
Can you add some comment to describe what this does?

Reply:
yes sure, two params in array and some dapps mixed up params, we need to find which one is address and which one is message

Reply:
I mean a comment in the code :)